lib/misc/blocks.js

```diff
@@ -46,13 +46,12 @@ function walkForward (ed, start) {
   while (mark < ed.getLastBufferRow()) {
     mark++
     const lineInfo = getLine(ed, mark)
+
     if (isStart(lineInfo)) {
       break
     }
     if (isEnd(lineInfo)) {
-      if (!(forLines(ed, start, mark-1).length === 0)) {
-        end = mark
-      }
+      end = mark
     } else if (!(isBlank(lineInfo) || isStart(lineInfo))) {
```

Comment:
nit: we can remove the second `isStart` condition since it's always `false` in this path.